vignettes/plotting_vignette.Rmd

```diff
@@ -0,0 +1,181 @@
+
+
+basic_plot + 
+  # Cutoff Band - a transparent rectangle
+  geom_rect(aes(xmin = 0,xmax = 30,ymin = -cutoff,ymax = cutoff),
+            alpha = 0.1,fill = "skyblue") +
+  # Titles
+  ggtitle(
+    label = paste("Best Model Fit",
+                  out[, "name"],
+                  sep = "\n"),
+    subtitle = paste("Assay Endpoint: ",
+                     out[, "assay"])) +
+  ## Add BMD and BMR labels
+  geom_hline(
+    aes(yintercept = out[, "bmr"]),
+    col = "blue") +
+  geom_segment(
+    aes(x = out[, "bmd"], xend = out[, "bmd"], y = -0.5, yend = out[, "bmr"]),
+    col = "blue"
+  ) + geom_point(aes(x = out[, "bmd"], y = out[, "bmr"], fill = "BMD"), shape = 21, cex = 2.5)
+```
+
+**Figure 3**: Basic plot generated with `concRespPlot2` with updated titles to provide additional details about the observed data. Experimental data is shown with the open circles and the red curve represents the best fit model. The title and subtitle display the compound name and assay endpoint, respectively. The light blue band represents responses within the cutoff threshold(s) -- i.e. cutoff band. The red point represents the BMD estimated from the winning model, given the BMR.  The blue segments display the BMR and the estimated BMD (horizontal and vertical segments, respectively).
+
+## Inclusion of Potency Estimates
+`concRespCore`, and `tcplfit2_core`, returns several potency estimates in addition to the BMD (displayed in Figure 3) - e.g. AC50, ACC, etc. Users may want to compare several potency estimates on the plot.  The code chunk below demonstrates how to add all available potency estimates from to the base plot. Note, when adding potencies to a plot where `log_conc = TRUE` is in use, the potency values also need to be log-transformed to be displayed in the correct positions.
+```{r}
+# Get all potency estimates and the corresponding y value on the curve
+estimate_points <- out %>%
+  select(bmd, acc, ac50, ac10, ac5) %>%
+  tidyr::pivot_longer(everything(), names_to = "Potency Estimates") %>%
+  mutate(`Potency Estimates` = toupper(`Potency Estimates`)) 
+
+y = c(out[, "bmr"], out[, "cutoff"], rep(out[, "top"], 3))
+y = y * c(1, 1, .5, .1, .05)
+estimate_points <- cbind(estimate_points, y = y)
+
+# add Potency Estimate Points and set colors
+basic_plot + geom_point(
+  data = estimate_points,
+  aes(x = value, y = y, fill = `Potency Estimates`), shape = 21, cex = 2.5
+)
+```
+
+**Figure 4**: Basic plot generated by `concRespPlot2` with potency estimates highlighted. Experimental data is shown with the open circles and the red curve represents the best fit model. Five colored points represent the various potency estimates from `concRespCore`.  These include the activity concentrations at 5, 10, and 50 percent of the maximal response (AC5 = gold, AC10 = red, and AC50 = green, respectively), as well as the activity concentration at the user-specified threshold (cutoff) and BMD (ACC = blue and BMD = purple, respectively).
+
+```{r}
```

Comment:
Please double check this figure.  Looked like the potency estimates were not actually aligning with the curve fit when the concentration was transformed to the log-scale.